core/boot/src/main/java/org/eclipse/dataspaceconnector/boot/system/DefaultServiceExtensionContext.java

```diff
@@ -47,28 +48,14 @@ public DefaultServiceExtensionContext(TypeManager typeManager, Monitor monitor,
         registerService(TypeManager.class, typeManager);
         registerService(Monitor.class, monitor);
         registerService(Telemetry.class, telemetry);
+        registerService(Clock.class, Clock.systemUTC());
```

Comment:
Unless there is a good reason to the contrary we should provide the clock with a `@Provider(isDefault=true)`-annotated factory method in the `DefaultServicesExtension`.

Reply:
Great input, done :)

Reply:
The reasons for this are:
- to provide a utility method `ServiceExtensionContext#getClock` so that extensions don't need to inject a clock service. If we move the default clock to `DefaultServicesExtension`, we can't have such a utility method, since we need the dependency resolver to start extensions in the right order.
- that the clock is a common service just like `TypeManager`, `Monitor` and `Telemetry`, and there's no obvious reason to manage it differently